quill-core/src/main/scala/io/getquill/idiom/ReifyStatement.scala

```diff
@@ -12,19 +14,33 @@ object ReifyStatement {
     statement:             Statement,
     forProbing:            Boolean
   ): (String, List[ScalarLift]) = {
-    def apply(acc: (String, List[ScalarLift]), token: Token): (String, List[ScalarLift]) =
-      (acc, token) match {
-        case ((s1, liftings), StringToken(s2))            => (s"$s1$s2", liftings)
-        case ((s1, liftings), Statement(tokens))          => tokens.foldLeft((s1, liftings))(apply)
-        case ((s1, liftings), ScalarLiftToken(lift))      => (s"$s1${liftingPlaceholder(liftings.size)}", liftings :+ lift)
-        case ((s1, liftings), SetContainsToken(a, op, b)) => apply(s1 -> liftings, stmt"$a $op ($b)")
-      }
     val expanded =
       forProbing match {
         case true  => statement
         case false => expandLiftings(statement, emptySetContainsToken)
       }
-    apply(("", List.empty), expanded)
+    token2sql(expanded, liftingPlaceholder)
+  }
+
+  private def token2sql(token: Token, liftingPlaceholder: Int => String): (String, List[ScalarLift]) = {
+    @tailrec
```

Comment:
Can we could simply convert the `List` to `Vector`  to solve `append` performance